runelite-client/src/main/java/net/runelite/client/plugins/agility/CrossbowmanStatue.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2018, Pratted <https://github.com/Pratted>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.agility;
+
+import com.google.common.base.Stopwatch;
+import lombok.Getter;
+import net.runelite.api.GameObject;
+import net.runelite.api.Renderable;
+
+import java.time.Duration;
+
+// There is no easy way to determine if a statue in animating so this
+// so this implementation uses the statue's renderable to determine if it
+// is animating. If the renderable changes, we know that a new frame in
```

Comment:
I'm pretty sure this isn't correct? I think every frame uses the same renderable. The renderable changes when they send the command to play a new animation. This seems to match up with the code you have below. I also think this is all that really needs to be explained in the comment.

Reply:
I agree with you for the most part, although I have a couple points I would like to bring up:

1. What I was trying to say is that when the reference to the `Renderable` changes, the statue visibly changes. I'm assuming that when this happens, the statue is animating. I think we're both on the same page here.

2. Generally speaking, an animation is composed of frames where each frame is a static image. Whenever some graphical changes occur in animation, a new frame must be being displayed because each frame is a static image. When I noticed that the renderable changing caused a graphical change to the statue, naturally it seemed like a new frame of the animation must have been displayed because of the reasoning above. However, if every frame uses the same renderable like you said, then I agree that won't be the case. I just find it counterintuitive that every frame would use the same renderable because typically at least some frames (images) in an animation would have a slight graphical difference between one another. 

As long as you agree with 1, I think we're good to go and I'll revise the comment to be a lot more concise. I brought up point 2 because your reply seemed counterintuitive to me, but I trust your judgement.